### PR TITLE
Fix buffer overrun in NT_GNU_PROPERTY_TYPE_0 parser

### PR DESCRIPTION
**Pull request by martijnthe**
The `iter_notes` method has code to parse `NT_GNU_PROPERTY_TYPE_0` type notes. The contents of the note are interpreted as an array of `elffile.structs.Elf_Prop`s.

There was a bug where it would keep on parsing from the stream until the end of the *segment or section*. This is only correct if the note would be the last in the segment/section. In general, it should stop parsing until it reaches the end of the note's data buffer.

This PR fixes this bug.

Fixes: https://github.com/eliben/pyelftools/issues/534

### CODE REVIEW

**Comment by eliben** [comment]
@sevaa

**Comment by sevaa** [comment]
No objections. Compilers are a diverse bunch...

File: elftools/elf/notes.py [hunk 1]
```diff
@@ -49,7 +49,8 @@ def iter_notes(elffile, offset, size):
         elif note['n_type'] == 'NT_GNU_PROPERTY_TYPE_0':
             off = offset
             props = []
-            while off < end:
+            current_note_end = offset + note['n_descsz']
```
Comment (eliben) [inline elftools/elf/notes.py:52]: Can you add a small comment here explaining the semantics (the usage of `n_descsz`)? A link to some spec-like thing would be great too, if possible

**Review by eliben — CHANGES_REQUESTED** [review]
Thanks for the fix and the thorough testing added. Just one small request

**Comment by martijnthe** [comment]
@eliben I've addressed your comment and resolved the conflicts with the other PR.

**Review by eliben — APPROVED** [review]

**Pull request merged**